include/mbedtls/ccm.h

```diff
@@ -40,6 +42,106 @@ typedef struct {
 }
 mbedtls_ccm_context;
 
+#else /* MBEDTLS_CCM_ALT */
+#include "ccm_alt.h"
+
+typedef struct {
+    union
+    {
+        mbedtls_ccm_alt_context_t ccm_alt_ctx;    /*!< opaque context structure to be defined in ccm_alt.h */
+        mbedtls_cipher_context_t cipher_ctx;
+    }u_ctx;
+    unsigned int    is_alternative;
+}
```

Comment:
This demands the customer's `mbedtls_ccm_alt_context_t` declaration to have a copy of `mbedtls_cipher_context_t` at the beginning, doesn't it? Otherwise, e.g., it seems that `mbedtls_ccm_free` below might fail at the freeing operation `mbedtls_cipher_free( cipher_ctx );` because it tries to interpret an instance of `mbedtls_ccm_alt_context_t` as one of `mbetdtls_cipher_context_t`.
If yes, can we find a solution which does not put such a restriction on the alternative implementation?